pages/en/lb4/Decorator.md

```diff
@@ -0,0 +1,196 @@
+
+  ```ts
+    class MyController {
+      @get('/')
+      @param(offsetSpec)
+      @param(pageSizeSpec)
+      list(offset?: number, pageSize?: number) {}
+    }
+  ```
+
+  or
+  
+  ```ts
+    class MyController {
+      @get('/')
+      list(
+        @param(offsetSpec) offset?: number,
+        @param(pageSizeSpec) pageSize?: number,
+      ) {}
+    }
+  ```
+  
+  More example can be found in [Writing Controller methods](controller.htm#Writing-Controller-methods)
+  
+## Dependency Injection
+
+@inject is a decorator to annotate method arguments for automatic injection by LoopBack IoC container. It can only be applied on a **method parameter** or a **property**.
+
+The `@inject` decorator allows you inject dependencies bound to the application context, or
+to any implementation of the [Context](#context) object. You can bind values, class definitions and 
+provider functions to those contexts and then resolve values (or the results of functions that return those 
+values!) in other areas of your code.
+
+```ts
+// application.ts
+import {Application} from '@loopback/core';
+import 'fs-extra';
+class MyApp extends Application {
+  constructor() {
+    super();
+    const app = this;
+    const widgetConf = JSON.parse(fs.readSync('./widget-config.json'));
+    app.bind('config.widget').to(widgetConf);
+  }
+}
+```
+Now that we've bound the 'config.widget' key to our configuration object, we can inject it in our WidgetController:
+
+```ts
+// widget-controller.ts
+import {widgetSpec} from '../apispec';
+@api(widgetSpec)
+class WidgetController {
+  @inject('config.widget')
+  protected config: JSON; // This will be resolved at runtime!
+
+  // etc...
+}
+```
+
+It's been documented as a LoopBack core concept in [Dependency Injection](Dependency-Injection.htm)
+For more information, see the [Dependency Injection](Dependency-Injection.htm) section under [LoopBack Core Concepts](Concepts.htm)
+
+
+## Authentication Decorator
+
+- @authenticate
+
+  Mark a controller method as requiring authenticated user, it takes in a strategy name. An example is using BasicStrategy to authenticate user in function `whoAmI`:
+
+  ```ts
+  // my-controller.ts
+  import { authenticate } from '@loopback/authentication';
+  import { inject } from '@loopback/context';
+
+  class MyController {
+    constructor(
+      @inject(BindingKeys.Authentication.CURRENT_USER)
+      private user: UserProfile,
+    ) {}
+
+    @authenticate('BasicStrategy')
+    async whoAmI() : Promise<string> {
+      return this.user.id;
+    }
+  }
+  ```
+
+## Repository Decorators
+
+### Model decorators
+
+- @model
+
+  Define a model in a repository, see [define-models](http://loopback.io/doc/en/lb4/Repositories.html#define-models)
+
+- @property
+
+  Define a property in a model, see [define-models](http://loopback.io/doc/en/lb4/Repositories.html#define-models)
+
+### Relation decorators
+
+  *Since relation is still in design phase, examples TBD.*
```

Comment:
Even the placeholders need formatting.
"This feature has not yet been released in alpha form. Documentation will be added here as this feature progresses."